packages/plugin/src/rules/naming-convention.ts

```diff
@@ -225,61 +283,67 @@ const rule: GraphQLESLintRule<NamingConventionRuleConfig> = {
       ObjectTypeDefinition: node => {
         if (options.ObjectTypeDefinition) {
           const property = normalisePropertyOption(options.ObjectTypeDefinition);
-          checkNode(node.name, property.style, 'Type', property.suffix, property.prefix);
+          checkNode(node.name, property, 'Type');
         }
       },
       InterfaceTypeDefinition: node => {
         if (options.InterfaceTypeDefinition) {
           const property = normalisePropertyOption(options.InterfaceTypeDefinition);
-          checkNode(node.name, property.style, 'Interface', property.suffix, property.prefix);
+          checkNode(node.name, property, 'Interface');
         }
       },
       EnumTypeDefinition: node => {
         if (options.EnumTypeDefinition) {
           const property = normalisePropertyOption(options.EnumTypeDefinition);
-          checkNode(node.name, property.style, 'Enumerator', property.suffix, property.prefix);
+          checkNode(node.name, property, 'Enumerator');
         }
       },
       InputObjectTypeDefinition: node => {
         if (options.InputObjectTypeDefinition) {
           const property = normalisePropertyOption(options.InputObjectTypeDefinition);
-          checkNode(node.name, property.style, 'Input type', property.suffix, property.prefix);
+          checkNode(node.name, property, 'Input type');
         }
       },
-      FieldDefinition: node => {
-        if (options.FieldDefinition) {
+      FieldDefinition: (node: any) => {
+        // TODO: check proper node typing
+        if (options.QueryDefinition && isQueryType(node.parent)) {
+          const property = normalisePropertyOption(options.QueryDefinition);
+          checkNode(node.name, property, 'Query');
+        }
+
+        if (options.FieldDefinition && !isQueryType(node.parent)) {
           const property = normalisePropertyOption(options.FieldDefinition);
-          checkNode(node.name, property.style, 'Field', property.suffix, property.prefix);
+          checkNode(node.name, property, 'Field');
```

Comment:
As mentioned in the description, this could be a breaking change because `FieldDefinition` no longer targets queries

Reply:
That's fine :) I think it's a fair change. We accept breaking changes until v1, so it's not a big deal at the moment.